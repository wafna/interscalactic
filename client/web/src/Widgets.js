import React from 'react';
import {orElse, check} from './Util';
export const icons = {
  CircleX: <Icon name='circle-x'/>
};
export function Icon(props) {
  let size = orElse(props.size, {height: 24, width: 24});
  // Could check this in PropType but we also need to transform.
  const extent = (() => {
    if (check.isNumber(props.size)) {
      return {height: size, width: size};
    } else if (check.isObject(size)) {
      return size;
    } else {
      throw new Error('Number or {height, width} required.');
    }
  })();
  return <img className={orElse(props.className)}
              src={'/open-iconic/svg/' + props.name + '.svg'}
              alt={orElse(props.alt, props.name)}
              height={extent.height} width={extent.width}
              onClick={e => {
                e.preventDefault();
                props.onClick && props.onClick()
              }}/>
}
export function Input(props) {
  return <input type="text" value={props.lens.take()} onChange={e => {
    props.lens.put(e.target.value);
    e.preventDefault();
  }}/>;
}
export function Button(props) {
  return <button disabled={props.disabled} onClick={e => {
    props.onClick();
    e.preventDefault();
  }}>{props.children}</button>
}
