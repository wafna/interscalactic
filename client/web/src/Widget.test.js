import React from 'react';
import renderer from 'react-test-renderer';
import {Icon} from './Widgets';
test('icon sets default height and width', () => {
  const icon = renderer.create(<Icon name='aperture'/>);
  let tree = icon.toJSON();
  expect(tree.props.height).toBe(24);
  expect(tree.props.width).toBe(24);
  expect(tree).toMatchSnapshot();
});
test('icon sets height and width from size as object', () => {
  const icon = renderer.create(<Icon name='aperture' size={{height: 80, width: 80}}/>);
  let tree = icon.toJSON();
  expect(tree.props.height).toBe(80);
  expect(tree.props.width).toBe(80);
});
test('icon sets height and width from size as number', () => {
  const icon = renderer.create(<Icon name='aperture' size={42}/>);
  let tree = icon.toJSON();
  expect(tree.props.height).toBe(42);
  expect(tree.props.width).toBe(42);
});
