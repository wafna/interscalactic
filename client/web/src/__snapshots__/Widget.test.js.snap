// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon sets default height and width 1`] = `
<img
  alt="aperture"
  className={null}
  height={24}
  onClick={[Function]}
  src="/open-iconic/svg/aperture.svg"
  width={24}
/>
`;
